src/mceditlib/operations/analyze.py

```diff
@@ -0,0 +1,87 @@
+"""
+   block_fill.py
+
+   Optimized functions for mass-replacing blocks in a world.
+"""
+from __future__ import absolute_import
+import logging
+
+import numpy
+from collections import defaultdict
+from mceditlib.operations import Operation
+
+log = logging.getLogger(__name__)
+
+
+
+class AnalyzeOperation(Operation):
+    def __init__(self, dimension, selection):
+        """
+        Analyze all blocks in a selection.
+
+        If blocksToReplace is given, it may be a list or tuple of blocktypes to replace with the given blocktype.
+
+        Additionally, blockType may be given as a list of (oldBlockType, newBlockType) pairs
+        to perform multiple replacements.
+
+        If updateLights is True, also checks to see if block changes require lighting updates and performs them.
+
+        :type dimension: WorldEditorDimension
+        :type selection: `~.BoundingBox`
+        """
+        super(AnalyzeOperation, self).__init__(dimension, selection)
+
+        self.createSections = False
+        self.blocks = numpy.zeros(65536, dtype='uint32')
+        self.selection = selection
+        self.entityCounts = defaultdict(int)
+        self.tileEntityCounts = defaultdict(int)
+        
+        self.chunkCount = 0
+        self.skipped = 0
+        self.sections = 0
+        log.info("Analyzing %s blocks", selection.volume)
+
+    def done(self):
+        log.info(u"Analyze: Skipped {0}/{1} sections".format(self.skipped, self.sections))
+        self.dimension.worldEditor.analyzeBlockOutput = self.blocks
```

Comment:
There's no point in shoving the output into worldEditor. It's fine where it is on AnalyzeOperation

If you want to cache it somewhere, AnalyzeOutputDialog would be a better place. You'd want to create the dialog when the editorsession is created and connect it to editorSession.revisionChanged to dump the cache.